kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/kubectl/equivalents/ConfigListUsers.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.examples.kubectl.equivalents;
+
+import io.fabric8.kubernetes.api.model.Config;
+import io.fabric8.kubernetes.api.model.NamedAuthInfo;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+/**
+ * Java equivalent to `kubectl config view -o jsonpath='{.users[*].name}'`. Gets a list of users
+ * found in KubeConfig
+ */
+public class ConfigListUsers {
+  private static final Logger logger = LoggerFactory.getLogger(ConfigListUsers.class);
+
+  public static void main(String[] args) throws FileNotFoundException {
+    // Gets KubeConfig via reading KubeConfig environment variable or ~/.kube/config
+    String kubeConfigFileName = io.fabric8.kubernetes.client.Config.getKubeconfigFilename();
+    if (kubeConfigFileName != null) {
+      File kubeConfigFile = new File(kubeConfigFileName);
+      Config kubeConfigObj = Serialization.unmarshal(new FileInputStream(kubeConfigFile), Config.class);
```

Comment:
I recall we exposed the applicable config, what's missing so that we don't have to deserialize the config file again?

Reply:
Do you mean `Config.getKubeConfigContents()` which got added in https://github.com/fabric8io/kubernetes-client/pull/2290? I'll check if it works when I update it

Reply:
Yes, I remember Andre provided some refactors to expose some configuration fields. So not sure if these values can or cannot be picked up from there and why.

Reply:
I checked `Config` but I could not find a method which exposes list of users. It exposes stuff like `NamedContext` all and current ones but not users.

Reply:
So does it make sense to store this information there?

What I mean is that we are already adding/caching the contexts (#2290) in the config. A context without the rest of user info is kind of incomplete IMO.

Or maybe we show even store the whole `io.fabric8.kubernetes.api.model.Config` model. Remove the duplicate fields from Config (i.e. List<Context>) add extra getters and modify the current getContexts() to retrieve values from the local `io.fabric8.kubernetes.api.model.Config`.

Reply:
I think we should do it. But in the scope of another issue.